client/src/app/core/services/http.service.ts

```diff
@@ -1,105 +1,92 @@
 import { Injectable } from '@angular/core';
 import { HttpClient } from '@angular/common/http';
-import { Observable } from 'rxjs';
-import { tap } from 'rxjs/operators';
 
 /**
  * Enum for different HTTPMethods
  */
 export enum HTTPMethod {
-    PUT,
-    PATCH
+    GET = 'get',
+    POST = 'post',
+    PUT = 'put',
+    PATCH = 'patch',
+    DELETE = 'delete'
 }
 
+/**
+ * Service for managing HTTP requests. Allows to send data for every method. Also (TODO) will do generic error handling.
+ */
 @Injectable({
     providedIn: 'root'
 })
-/**
- * Service for sending data back to server
- */
+
 export class HttpService {
     /**
-     * Construct a DataSendService
+     * Construct a HttpService
      *
      * @param http The HTTP Client
      */
     public constructor(private http: HttpClient) {}
 
+    public async send<T>(url: string, method: HTTPMethod, data?: any): Promise<T> {
```

Comment:
why public if you have meta-functions for all http-methods?

Reply:
This should `<void>` or `id`

Reply:
The http service is used in more cases then just models, so there must be more return types.